protocols/kad/src/behaviour.rs

```diff
@@ -1054,12 +1049,14 @@ where
     }
 
     fn inject_connected(&mut self, peer: PeerId, endpoint: ConnectedPoint) {
-        while let Some(pos) = self.pending_rpcs.iter().position(|(p, _)| p == &peer) {
-            let (_, rpc) = self.pending_rpcs.remove(pos);
-            self.queued_events.push_back(NetworkBehaviourAction::SendEvent {
-                peer_id: peer.clone(),
-                event: rpc,
-            });
+        // Queue events for sending pending RPCs to the connected peer.
+        // There can be only one pending RPC for a particular peer and query per definition.
+        for (peer_id, event) in self.queries.iter_mut().filter_map(|q|
```

Comment:
I'm not a fan of these complicated nested functional-style getters, but it's also not a big deal.